pkg/jx/cmd/test_data/step_create_task/script-step/jenkins-x.yml

```diff
@@ -0,0 +1,38 @@
+pipelineConfig:
+  env:
+    - name: FRUIT
+      value: BANANA
+  pipelines:
+    release:
+      pipeline:
+        env:
+          - name: GIT_AUTHOR_NAME
+            value: somebodyelse
+        options:
+          containerOptions:
+            resources:
+              requests:
+                cpu: 0.1
+                memory: 64Mi
+        agent:
+          image: nodejs
+        stages:
+          - name: Build-a-really-long-stage-name-please-but-not-too-long-thanks
+            steps:
+              - script: |
```

Comment:
My understanding is that the shell used to execute the script defaults to `/bin/sh` but could be overridden by the pod template. I wonder if people would want to be able to specify a different shell here since it would be so easy to write scripts directly in jenkins-x.yml. You could still do that in `script` as-is, but it would be a bit awkward with quotes. 

That said, maybe forcing `/bin/sh` as a default is a good way to encourage people to put their scripts in source control somewhere if they are complicated enough to want a different shell.

Reply:
Yeah, I'd say stick with forcing the shell - if someone really cares enough, they can hack the `script`, but it's not worth the potential pain of someone specifying an overriding shell that doesn't exist on the image, IMO.